### ISSUE TITLE
Upgrade CRDs to apiextensions.k8s.io/v1

### ISSUE THREAD

**Issue by wreed4** [argoproj/argo#4765]
# Summary

The CRD apiVersion needs to be upgraded for 1.16.  Having them on the old version breaks Helm linting:
```
==> Linting .
[INFO] Chart.yaml: icon is recommended
[ERROR] templates/argorollouts_v0.10.1.yaml: the kind "apiextensions.k8s.io/v1beta1 CustomResourceDefinition" is deprecated in favor of "apiextensions.k8s.io/v1 CustomResourceDefinition"
[ERROR] templates/argoworkflows_v2.11.8.yaml: the kind "apiextensions.k8s.io/v1beta1 CustomResourceDefinition" is deprecated in favor of "apiextensions.k8s.io/v1 CustomResourceDefinition"
```

# Use Cases

Normally this doesn't matter, but helm linting breaking prevents us from using these in our helm charts as we want to lint our other resources.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by alexec** [comment]
We're using `controller-tools v0.2.9` and I think we should upgrade to v0.4.1. Would you like to submit a PR?

`v1beta1` will be deprecated in Kubernetes 1.22, but Helm is important too.

**Comment by JPZ13** [comment]
@alexec I can take this. I was just in the `hack` folder with #4810 and upgraded `controller-tools` to v0.4.1. Should be a relatively quick change

**Comment by JPZ13** [comment]
After diving into this, I think it is blocked by the steps issue from #4828 that I reverted on #4810. To generate v1 CRDs successfully, I think we'll have to fix the inlining issue in apimachinery. Here's the error message I get when running `make install`:

```
The CustomResourceDefinition "workflows.argoproj.io" is invalid: metadata.annotations: Too long: must have at most 262144 bytes
```

And when running `kubectl -n argo create -l app.kubernetes.io/part-of=argo -f dist/minimal.yaml`, the pods don't start. I've pushed my latest commits to #4849 as a reference

cc @alexec @taharah
